src/logging/NewRelicLoggingService.js

```diff
@@ -77,14 +112,26 @@ export default class NewRelicLoggingService {
       // noticeError expects undefined if there are no custom attributes.
       allCustomAttributes = undefined;
     }
-    /* istanbul ignore next */
-    if (process.env.NODE_ENV === 'development') {
-      console.error(error, allCustomAttributes); // eslint-disable-line
-    }
-    /* istanbul ignore else */
-    if (window && typeof window.newrelic !== 'undefined') {
-      // Note: customProperties are not sent.  Presumably High-Security Mode is being used.
-      window.newrelic.noticeError(fixErrorLength(error), allCustomAttributes);
+
+    /*
+        Separate the errors into expected errors and unexpected errors.
+        Expected errors are logged via adding a page action.
+        Unexpected errors are logged via noticeError and count as "JS Errors" for the application.
+    */
+    /* for (const expectedErr of this.expectedErrors) { */
+    let expected = false;
+    this.expectedErrors.forEach(expectedErr => {
+      if (!expected
+          && ((typeof error === 'string' && error.includes(expectedErr))
+          || (error.message && error.message.includes(expectedErr)))) {
```

Comment:
Currently, this code only supports substring detection. Should it instead support regex expressions?

Reply:
I think regex would be good, especially since one the expected errors messages is literally `"Script error"` -- wouldn't want to match every message with that substring!

Reply:
JS tip -- [Array.some](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/some) is a nice way to do the check-against-event-array-element loop idiom here.